libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -437,16 +437,22 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 		type = TypeProvider::withLocation(ref, typeLoc, isPointer);
 	}
 
-	if (
-		_variable.isConstant() &&
-		!dynamic_cast<UserDefinedValueType const*>(type) &&
-		type->containsNestedMapping()
-	)
-		m_errorReporter.fatalDeclarationError(
-			3530_error,
-			_variable.location(),
-			"The type contains a (nested) mapping and therefore cannot be a constant."
+	if (_variable.isConstant())
+	{
+		if (!type->isValueType())
+		{
+			bool allowed = false;
+			if (auto arrayType = dynamic_cast<ArrayType const*>(type))
+				allowed = arrayType->isByteArray();
+			if (!allowed)
+				m_errorReporter.fatalTypeError(9259_error, _variable.location(), "Constants of non-value type not yet implemented.");
```

Comment:
Actually, since we allow byte arrays, the error message is (and has been) wrong :-). But I guess we can keep it.
You can also change it to sth like "Only constants of value types and byte array types are implemented." or so.